plugins/modules/ntnx_clusters.py

```diff
@@ -0,0 +1,522 @@
+            default_vswitch_config:
+                description:
+                - default vswitch configuration
+                type: dict
+                suboptions:
+                    nic_teaming_policy:
+                        type: str
+                        description:
+                        - nice teaming policy
+                    uplink_grouping:
+                        type: str
+                        description:
+                        - determines how the ethernet uplinks are selected for this vswitch
+
+extends_documentation_fragment:
+- nutanix.ncp.ntnx_credentials
+- nutanix.ncp.ntnx_operations
+author:
+- Thomas Tomlinson (@thomas-tomlinson)
+"""
+
+EXAMPLES = r"""
+
+- name: ensure public ssh keys are present
+  nutanix.ncp.ntnx_clusters:
+    validate_certs: False
+    state: present
+    nutanix_host: nutanix_host
+    nutanix_username: nutanix_user
+    nutanix_password: nutanix_password
+    cluster:
+        name: cluster_name
+    authorized_public_key_list:
+        - name: user_1
+          key: ssh-rsa REALLY_LONG_KEY_TEXT
+        - name: user_2
+          key: ssh-rsa REALLY_LONG_KEY_TEXT
+
+"""
+
+RETURN = r"""
+response:
+    description:
+    - The full API response from the cluster PUT call.
+    returned: always
+    type: dict
+    sample:  {
+        "api_version": "3.1",
+            "metadata": {
+                "categories": {},
+                "categories_mapping": {},
+                "creation_time": "2022-05-27T15:31:23Z",
+                "kind": "cluster",
+                "last_update_time": "2022-06-02T19:15:48Z",
+                "owner_reference": {
+                    "kind": "user",
+                    "name": "admin",
+                    "uuid": "00000000-0000-0000-0000-000000000000"
+                },
+                "spec_version": 51,
+                "uuid": "0005dec1-a7d4-a09e-3740-2cea7f9c9e08"
+            },
+            "spec": {
+                "name": "cluster_name",
+                "resources": {
+                    "config": {
+                        "authorized_public_key_list": []
+                    }
+                }
+            }
+        }
+"""
+
+from ..module_utils.base_module import BaseModule  # noqa: E402
+from ..module_utils.prism.clusters import Cluster  # noqa: E402
+from ..module_utils.prism.tasks import Task  # noqa: E402
+from ..module_utils.utils import remove_param_with_none_value  # noqa: E402
+
+
+def get_module_spec():
+    mutually_exclusive = [("name", "uuid")]
+
+    entity_by_spec = dict(name=dict(type="str"), uuid=dict(type="str"))
+
+    domain_server_spec = dict(
+        name=dict(type="str"),
+        nameserver=dict(type="str"),
+        domain_credentials=dict(
+            type="dict", options=dict(
+                username=dict(type="str"),
+                password=dict(type="str", no_log=True),
+            )
+        )
+    )
+
+    http_proxy_list_spec = dict(
```

Comment:
Incase some field is required in subspec of http_proxy_list_spec, domain_server_spec, etc. mark it as required=True